packages/sdk/src/modules/moduleFromEndpoints/types.ts

```diff
@@ -134,6 +166,32 @@ export type Options = {
    * Default request config for each request.
    */
   defaultRequestConfig?: IncomingConfig;
+
+  /**
+   * An optional custom error handler for HTTP requests.
+   *
+   * If not provided, errors will be thrown as is.
+   *
+   * This enables custom error handling, like retrying the request or refreshing tokens, depending on the error type and details of the request that failed.
+   *
+   * @example
+   * ```typescript
+   * const options: Options = {
+   *   apiUrl: "https://api.example.com",
+   *   errorHandler: async ({ error, url, params, config, httpClient }) => {
+   *     if (error.status === 401) {
+   *       // Refresh token
+   *       await refreshToken();
+   *       // Retry the request
+   *       return httpClient(url, params, config);
+   *     }
```

Comment:
I think it might be quite common to do conditions basing on the method name - e.g.

```suggestion
   *     const methodName = url.split('/').at(-1);
   *     if (error.status === 401 && methodName !== "loginCustomer") {
   *       // Refresh token
   *       await refreshToken();
   *       // Retry the request
   *       return httpClient(url, params, config);
   *     }
```

hence I think it would be more convenient to pass also `methodName` in `ErrorHandlerContext`, but leaving it up to you

Reply:
Done